### DISCUSSION TITLE
How to achieve finite volume in Moose？

### DISCUSSION THREAD

**Discussion by Moyiqi** [idaholab/moose#24516] [Q&A General] [answered]
Hello everyone, may I ask two questions?
I want to implement the finite volume method in moose, may I ask which modules need to be mobilized? 
And how to realize mapping for non-finite element discrete physical field in MOOSE?
Thank you very much！

Lily

**Comment by GiudGiud** [comment, marked as answer]
Hello

Finite volume is in the framework, it can be utilized in any module.
The navier stokes module is currently the only example using finite volume

Where is the non-finite element discrete field coming from? Does it have an analytical form?

Guillaume

**Reply by Moyiqi** [reply]
Sorry,I don't make myself clear enough. In other words，how does moose implement finite element to infinite element mapping?

Lily

**Reply by GiudGiud** [reply]
Not sure. Libmesh has a concept of infinite element you ll want to have a look at their website maybe ask on the libmesh discussions forum

**Reply by Moyiqi** [reply]
Thank you!
Lily

**Reply by lindsayad** [reply]
@roystgnr

**Reply by roystgnr** [reply]
I've never actually used the infinite elements capability myself.  It's got a bit of test coverage in libMesh that does double duty as usage examples: miscellaneous examples 1, 14, and 15 ... but has it ever been used in MOOSE?  It's a non-default configuration option and I don't think it's one that the default MOOSE libMesh build script has enabled ... and although MOOSE users can work with a custom-configured libMesh, `git grep InfElemBuilder` isn't showing me anything in MOOSE, and I think a meshgenerator calling on that would have been the very first thing a user would have added.  It wouldn't have been the last thing, though; the Sobolev-weighted quadrature needed for element assembly is slightly different too.

**Reply by Moyiqi** [reply]
Thank you for your answer. It's very enlightening.
Lily
